db/migrations/20190913165643-add-indexes-to-unfurls-table.js

```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  up: (queryInterface) => Promise.all([
+    queryInterface.addIndex('Unfurls', {
+      fields: ['channelSlackId', 'slackWorkspaceId', 'slackUserId', 'githubRepoId'],
```

Comment:
Awesome, thanks for getting this started ✨ 

I'm not 100% sure here, but if we add the index this way, is it a "composite" index that only ends up becoming helpful if we have a query that specifies all four fields in the `WHERE` clause?

I've previously seen indexes get added individually for each column, like in this migration: https://github.com/integrations/slack/blob/7d4c2d323396c06e7744665b5e008f241154d2f5/db/migrations/20180117192603-add-indexes.js#L4-L23

Reply:
@wilhelmklopp is exactly right. 

> A multicolumn B-tree index can be used with query conditions that involve any subset of the index's columns, but **the index is most efficient when there are constraints on the leading (leftmost) columns.** The exact rule is that equality constraints on leading columns, plus any inequality constraints on the first column that does not have an equality constraint, will be used to limit the portion of the index that is scanned. 
> – https://www.postgresql.org/docs/10/indexes-multicolumn.html

Unless we plan to query against all four columns in the same query or only the leftmost columns, splitting this into four, one-column indexes will provide the most flexibility while debugging.

I'm happy to chat about this further. Some of these column constraints can be difficult to explain without examples.